web/src/server/api/routers/comments.ts

```diff
@@ -0,0 +1,192 @@
+import { z } from "zod";
+
+import { throwIfNoProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import {
+  createTRPCRouter,
+  protectedProjectProcedure,
+} from "@/src/server/api/trpc";
+import { CommentObjectType } from "../../../../../packages/shared/dist/prisma/generated/types";
+import { Prisma } from "@langfuse/shared";
+
+const CreateCommentData = z.object({
+  projectId: z.string(),
+  content: z.string(),
+  objectId: z.string(),
+  objectType: z.nativeEnum(CommentObjectType),
+});
+
+const COMMENT_OBJECT_TYPE_TO_PRISMA_MODEL = new Map<CommentObjectType, string>([
+  [CommentObjectType.TRACE, "trace"],
+  [CommentObjectType.OBSERVATION, "observation"],
+  [CommentObjectType.SESSION, "session"],
+  [CommentObjectType.PROMPT, "prompt"],
+]);
+
+const validateCommentReferenceObject = async ({
+  ctx,
+  input,
+}: {
+  ctx: any;
+  input: z.infer<typeof CreateCommentData>;
+}): Promise<void> => {
+  const { objectId, objectType, projectId } = input;
+  const prismaModel = COMMENT_OBJECT_TYPE_TO_PRISMA_MODEL.get(
+    objectType,
+  ) as string;
+
+  const object = await ctx.prisma[prismaModel].findFirst({
+    where: {
+      id: objectId,
+      projectId,
+    },
+  });
+
+  if (!object) {
+    throw new Error(
+      `No ${prismaModel} with id ${objectId} in project ${projectId}`,
+    );
+  }
+};
+
+export const commentsRouter = createTRPCRouter({
```

Comment:
can you add try .. catch blocks within the routers? If you then log the message, it helps us to find errors in our logs. Thx!

Reply:
Of course, thx 👍